moped-editor/src/views/projects/projectView/ProjectComponents/index.js

```diff
@@ -108,8 +144,76 @@ export default function MapView({ projectName, projectStatuses }) {
   };
 
   const onSaveComponent = () => {
```

Comment:
as these callbacks continue to accumulate here in `index.js` - i'm wondering if/when we want to think about memoizing them with `useCallback`. my limited experience with that hook is that it can be a real chore to get the dependencies right.

i definitely think there are diminishing returns trying to optimize—whether it's the added complexity of `useCallback` or the dependency mess of extracting the function out of the component with a long list of params. 

taking a really quick look, this `onSaveComponent` actually might not be so bad if it were extracted? i trust your judgement on this @mddilley but wanted to throw that out there.

Reply:
Cool, I'll give it a whirl to see and agree that it could be good to wait to optimize if it is too burdensome. 🙏